Api/MyMods.cs

```diff
@@ -16,13 +16,19 @@ class MyMods {
         #region Event Handlers
 
         public static void SelectEnabled_chb_CheckedChanged(object sender, EventArgs e) {
-            foreach (Panel p in Program.MasterForm.modListPanel.Controls.OfType<Panel>()) {
-                CheckBox chb = (CheckBox)p.GetChildAtPoint(Config.MyModsChbPoint);
-                string modpackname = chb.Text.Replace(Config.dirtyPadding, "");
-                if (Config.IsPatched(modpackname)) {
-                    chb.Checked = ((CheckBox)sender).Checked;
-                }
-            }
+            ApplyBulkChecks(((CheckBox)sender).Checked, true);
+        }
+
+        public static void SelectDisabled_chb_CheckedChanged(object sender, EventArgs e) {
+            ApplyBulkChecks(((CheckBox)sender).Checked, false);
+        }
+
+        public static void SelectAll_chb_CheckedChanged(object sender, EventArgs e) {
+            Program.MasterForm.selectEnabled_chb.Checked = ((CheckBox)sender).Checked;
+            Program.MasterForm.selectDisabled_chb.Checked = ((CheckBox)sender).Checked;
+
+            ApplyBulkChecks(Program.MasterForm.selectEnabled_chb.Checked, true);
+            ApplyBulkChecks(Program.MasterForm.selectDisabled_chb.Checked, false);
```

Comment:
Instead of calling `ApplyBulkChecks()` twice, and therefore iterating over every listed modpack row twice, I would just copy the inner forloop from `ApplyBulkChecks()` and simply have it set every checkbox to `((CheckBox)sender).Checked`. This will make the function o(n) instead of o(2n)

Reply:
Sure thing. I made that in like, 10 mins. Done